app/gui/src/controller/searcher.rs

```diff
@@ -374,11 +369,55 @@ impl Searcher {
         self.breadcrumbs.names()
     }
 
-    /// Select the breadcrumb with the index [`id`]. The displayed content of the browser will be
-    /// updated.
+    /// Set the selected breadcrumb. The `id` is the index of the breadcrumb from left to right.
     pub fn select_breadcrumb(&self, id: usize) {
         self.breadcrumbs.select(id);
-        self.reload_list();
+    }
+
+    /// Set the breadcrumbs to match the component at the given `index`. The index refers to the
+    /// displayed list of components. Returns the full breadcrumb for the entry, if there is one.
+    pub fn update_breadcrumbs(&self, index: usize) -> Option<Vec<BreadcrumbEntry>> {
+        let data = self.data.borrow();
+        if let Some(component) = data.components.displayed().get(index) {
+            if let Some(id) = component.id() {
+                let bc_builder = breadcrumbs::Builder::new(&self.database);
+                let breadcrumbs = bc_builder.build(id).collect_vec();
+                assert!(breadcrumbs.iter().all(|e| self.database.lookup(e.id()).is_ok()));
+                self.breadcrumbs.set_content(breadcrumbs.clone().into_iter());
+                Some(breadcrumbs)
+            } else {
+                warn!(
+                    "Cannot update breadcrumbs with component that has no suggestion database \
+                entry. Invalid component: {:?}",
+                    component
+                );
+                None
+            }
+        } else {
+            warn!("Update readcrumbs called with invalid index: {}", index);
+            None
+        }
+    }
+
+    /// Return the full breadcrumb for the entry.
+    pub fn breadcrumbs_for_entry(&self, id: EntryId) -> Result<Vec<Breadcrumb>, NoSuchEntry> {
+        let component = self.database.lookup(id)?;
+        let name = &component.name;
+        let icon = Some(component.icon());
+        let module = &component.defined_in;
+        let breadcrumbs_base = module.path().iter().map(|name| Breadcrumb::new_without_icon(name));
```

Comment:
Here I see that we create breadcrumbs without icons for the modules path, but in the video, there are icons on each crumb. Is that correct?